docker/build-image.sh

```diff
@@ -56,39 +58,67 @@ while [[ $# -gt 0 ]]; do
             fi
             shift
             ;;
+        --focal)
+            BUILD_FOCAL=true
+            shift
+            ;;
+        --jammy)
+            BUILD_JAMMY=true
+            shift
+            ;;
     esac
 done
 
 if [[ -z "$COMPONENT_VERSION_STRING" ]]; then
     COMPONENT_VERSION_STRING=$("./get-component-version.sh")
 fi
 
-echo "Building docker image for $IMAGE version: $COMPONENT_VERSION_STRING"
-echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING"
+build_image() {
+    local dockerfile_path=$1
+    local tag_suffix=$2
 
-cd ..
-docker build --network=host --no-cache -t $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING \
-    --build-arg VERSION="$COMPONENT_VERSION_STRING" \
-    --build-arg VCS_REF=`git rev-parse --short HEAD` \
-    --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` .
+    echo "Building docker image for $IMAGE version: $COMPONENT_VERSION_STRING using Dockerfile: $dockerfile_path"
+    echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix"
+
+    DOCKER_BUILDKIT=1 docker build --network=host -t $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix \
+        --build-arg VERSION="$COMPONENT_VERSION_STRING" \
+        --build-arg VCS_REF=`git rev-parse --short HEAD` \
+        --build-arg BUILD_DATE=`date -u +"%Y-%m-%dT%H:%M:%SZ"` \
+        -f $dockerfile_path .
+
+    TAGS+=("$USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix")
+
+    docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix $USERNAME/$IMAGE:latest$tag_suffix
+    TAGS+=("$USERNAME/$IMAGE:latest$tag_suffix")
+
+    echo "Tagged $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix as $USERNAME/$IMAGE:latest$tag_suffix"
+}
 
 TAGS=()
-TAGS+=("$USERNAME/$IMAGE:$COMPONENT_VERSION_STRING")
 
-docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING $USERNAME/$IMAGE:latest
-TAGS+=("$USERNAME/$IMAGE:latest")
+cd ..
+
+# If neither --focal nor --jammy is specified, build only focal for now.
+# Once all humble related changes are merged into develop, jammy should be enabled.
+# https://usdot-carma.atlassian.net/browse/ARC-227
+if [ "$BUILD_FOCAL" = false ] && [ "$BUILD_JAMMY" = false ]; then
+    BUILD_FOCAL=true
+    BUILD_JAMMY=false
+fi
 
-echo "Tagged $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING as $USERNAME/$IMAGE:latest"
+# TODO, distinguish with suffix when Humble is fully integrated
+# until then focal will have no suffix and be the main image
+# https://usdot-carma.atlassian.net/browse/ARC-227
+if [ "$BUILD_FOCAL" = true ]; then
```

Comment:
Can we add echo "Building Focal Image..." and echo "Building Jammy Image..." some thing like that which is easy to debug during CI for each focal and jammy ?

Reply:
added